codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/customizations/HttpAuthDecorator.kt

```diff
@@ -0,0 +1,369 @@
+import software.amazon.smithy.rust.codegen.client.smithy.generators.config.ConfigCustomization
+import software.amazon.smithy.rust.codegen.client.smithy.generators.config.ServiceConfig
+import software.amazon.smithy.rust.codegen.core.rustlang.CargoDependency
+import software.amazon.smithy.rust.codegen.core.rustlang.Writable
+import software.amazon.smithy.rust.codegen.core.rustlang.rust
+import software.amazon.smithy.rust.codegen.core.rustlang.rustTemplate
+import software.amazon.smithy.rust.codegen.core.rustlang.withBlockTemplate
+import software.amazon.smithy.rust.codegen.core.rustlang.writable
+import software.amazon.smithy.rust.codegen.core.smithy.RuntimeConfig
+import software.amazon.smithy.rust.codegen.core.smithy.RuntimeType
+import software.amazon.smithy.rust.codegen.core.smithy.RustCrate
+import software.amazon.smithy.rust.codegen.core.util.dq
+import software.amazon.smithy.rust.codegen.core.util.getTrait
+import software.amazon.smithy.rust.codegen.core.util.letIf
+
+fun codegenScope(runtimeConfig: RuntimeConfig): Array<Pair<String, Any>> {
+    val smithyRuntime =
+        CargoDependency.smithyRuntime(runtimeConfig).withFeature("http-auth").toType()
+    val smithyRuntimeApi = CargoDependency.smithyRuntimeApi(runtimeConfig).withFeature("http-auth").toType()
+    return arrayOf(
+        "ApiKeyAuthScheme" to smithyRuntime.resolve("client::auth::http::ApiKeyAuthScheme"),
+        "ApiKeyLocation" to smithyRuntime.resolve("client::auth::http::ApiKeyLocation"),
+        "AuthOptionListResolver" to smithyRuntimeApi.resolve("client::auth::option_resolver::AuthOptionListResolver"),
+        "BasicAuthScheme" to smithyRuntime.resolve("client::auth::http::BasicAuthScheme"),
+        "BearerAuthScheme" to smithyRuntime.resolve("client::auth::http::BearerAuthScheme"),
+        "DigestAuthScheme" to smithyRuntime.resolve("client::auth::http::DigestAuthScheme"),
+        "HTTP_API_KEY_AUTH_SCHEME_ID" to smithyRuntimeApi.resolve("client::auth::http::HTTP_API_KEY_AUTH_SCHEME_ID"),
+        "HTTP_BASIC_AUTH_SCHEME_ID" to smithyRuntimeApi.resolve("client::auth::http::HTTP_BASIC_AUTH_SCHEME_ID"),
+        "HTTP_BEARER_AUTH_SCHEME_ID" to smithyRuntimeApi.resolve("client::auth::http::HTTP_BEARER_AUTH_SCHEME_ID"),
+        "HTTP_DIGEST_AUTH_SCHEME_ID" to smithyRuntimeApi.resolve("client::auth::http::HTTP_DIGEST_AUTH_SCHEME_ID"),
+        "HttpAuthOption" to smithyRuntimeApi.resolve("client::orchestrator::HttpAuthOption"),
+        "IdentityResolver" to smithyRuntimeApi.resolve("client::identity::IdentityResolver"),
+        "IdentityResolvers" to smithyRuntimeApi.resolve("client::identity::IdentityResolvers"),
+        "Login" to smithyRuntimeApi.resolve("client::identity::http::Login"),
+        "PropertyBag" to RuntimeType.smithyHttp(runtimeConfig).resolve("property_bag::PropertyBag"),
+        "Token" to smithyRuntimeApi.resolve("client::identity::http::Token"),
+    )
+}
```

Comment:
nit: `client::auth::http` is used so much that I think aliasing it would be nice here.
```suggestion
    val smithyRuntimeApi = CargoDependency.smithyRuntimeApi(runtimeConfig).withFeature("http-auth").toType()
    val httpAuth = smithyRuntime.resolve("client::auth::http")
    return arrayOf(
        "ApiKeyAuthScheme" to httpAuth.resolve("ApiKeyAuthScheme"),
        "ApiKeyLocation" to httpAuth.resolve("ApiKeyLocation"),
        "AuthOptionListResolver" to smithyRuntimeApi.resolve("client::auth::option_resolver::AuthOptionListResolver"),
        "BasicAuthScheme" to httpAuth.resolve("BasicAuthScheme"),
        "BearerAuthScheme" to httpAuth.resolve("BearerAuthScheme"),
        "DigestAuthScheme" to httpAuth.resolve("DigestAuthScheme"),
        "HTTP_API_KEY_AUTH_SCHEME_ID" to httpAuth.resolve("HTTP_API_KEY_AUTH_SCHEME_ID"),
        "HTTP_BASIC_AUTH_SCHEME_ID" to httpAuth.resolve("HTTP_BASIC_AUTH_SCHEME_ID"),
        "HTTP_BEARER_AUTH_SCHEME_ID" to httpAuth.resolve("HTTP_BEARER_AUTH_SCHEME_ID"),
        "HTTP_DIGEST_AUTH_SCHEME_ID" to httpAuth.resolve("HTTP_DIGEST_AUTH_SCHEME_ID"),
        "HttpAuthOption" to smithyRuntimeApi.resolve("client::orchestrator::HttpAuthOption"),
        "IdentityResolver" to smithyRuntimeApi.resolve("client::identity::IdentityResolver"),
        "IdentityResolvers" to smithyRuntimeApi.resolve("client::identity::IdentityResolvers"),
        "Login" to smithyRuntimeApi.resolve("client::identity::http::Login"),
        "PropertyBag" to RuntimeType.smithyHttp(runtimeConfig).resolve("property_bag::PropertyBag"),
        "Token" to smithyRuntimeApi.resolve("client::identity::http::Token"),
    )
}
```